client/src/components-page/profile/Profile.js

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect } from 'react';
+import { Link, withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { FaCaretRight } from 'react-icons/fa';
+
+import { getUserProfile } from '../../redux/profile/profile.actions';
+
+import Spinner from '../../components-ui/spinner/Spinner';
+import Button from '../../components-ui/button/Button';
+
+import style from './profile.module.scss';
+
+// *************************** PROFILE COMPONENT *************************** //
+const Profile = ({ currentProfile, loading, auth, match, getUserProfile }) => {
+  // 'match' passed down as prop from AppRouter
+  useEffect(() => {
+    getUserProfile(match.params.id);
+  }, [getUserProfile, match]);
+
+  return (
+    <div className={style.profile}>
+    {
+      currentProfile === null || loading
```

Comment:
I think this is written a bit awkwardly. Consider returning the div with the spinner in it before line 20, based on the conditional.

e.g. 
```
if (currentProfile === null || loading)
  return ( // stuff
else
  return ( // the profile
```

You will end up duplicating a bit but I think it reads cleaner.

Reply:
True yeah I agree, I'll take a look at that tomorrow